funnel/forms/proposal.py

```diff
@@ -74,10 +74,15 @@ def __init__(self, *args, **kwargs):
             self.description.description = space.proposal_part_b.get('hint')
 
 
-class ProposalStatusForm(forms.Form):
-    status = forms.SelectField(
-        __("Status"), coerce=int,
-        choices=[(status, label.title) for (status, label) in PROPOSALSTATUS.items() if status != PROPOSALSTATUS.DRAFT])
+class ProposalTransitionForm(forms.Form):
+    transition = forms.SelectField(__("Status"), validators=[forms.validators.DataRequired()])
+
+    def populate_transitions(self, proposal):
+        # value: transition method name
+        # label: method name in title case, after removing underscores
+        self.transition.choices = sorted([
+            (tname, tname.replace("_", " ").title()) for (tname, t) in proposal.state.transitions.iteritems()
```

Comment:
Use `iterkeys` since you're only using the key?

Reply:
@shreyas-satish Need a key and value for choices. And then wtforms uses this to validate the form.

Reply:
`(tname, tname.replace("_", " ").title()) for tname in proposal.state.transitions.iterkeys()` won't work?

Reply:
Oh right. yeah, that'll work.